nntrainer/graph/network_graph.cpp

```diff
@@ -577,15 +633,32 @@ NetworkGraph::getGraph(const std::string &input_layer,
 
   /** copy the graph and return */
   std::vector<std::shared_ptr<Layer>> ret;
-  std::copy(layers.begin() + num_layers_remove_start,
-            layers.end() - num_layers_remove_end, std::back_inserter(ret));
+  std::transform(adj.begin() + num_layers_remove_start,
+                 adj.end() - num_layers_remove_end, std::back_inserter(ret),
+                 [](auto const &elem) { return elem.front().layer; });
+
+  return ret;
+}
+
+std::vector<std::shared_ptr<Layer>> NetworkGraph::getLayers() {
+  std::vector<std::shared_ptr<Layer>> ret;
+  if (compiled) {
+    std::transform(Sorted.begin(), Sorted.end(), std::back_inserter(ret),
+                   [](auto const &elem) { return elem.layer; });
+  } else {
+    std::transform(adj.begin(), adj.end(), std::back_inserter(ret),
+                   [](auto const &elem) { return elem.front().layer; });
+  }
```

Comment:
I guess. once it is compiled.. then we do not need adj anymore. We could remove it.

Reply:
that's interesting, I will try and if it works, will add as a new commit.